### PR TITLE
Feature/aql subquery execute parallel gather 2

### PR DESCRIPTION
**Pull request by goedderz**
AQL execute internal API change for executors with multiple dependencies.

### CODE REVIEW

**Comment by goedderz** [comment]
http://jenkins.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8891/

**Comment by goedderz** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/8902/

File: arangod/Aql/AqlCall.cpp [hunk 1]
```diff
@@ -279,3 +280,19 @@ auto AqlCall::toString() const -> std::string {
   stream << *this;
   return stream.str();
 }
+
+auto aql::operator<<(std::ostream& out, AqlCall::Limit const& limit) -> std::ostream& {
```
Comment (mchacki) [inline arangod/Aql/AqlCall.cpp]: \o/

File: arangod/Aql/AqlCallStack.cpp [hunk 2]
```diff
@@ -87,6 +87,12 @@ void AqlCallStack::pushCall(AqlCall&& call) {
   _operations.push(call);
 }
 
+void AqlCallStack::pushCall(AqlCall const& call) {
+  // TODO is this correct on subqueries?
```
Comment (mchacki) [inline arangod/Aql/AqlCallStack.cpp]: The non-relevant stack will be removed.
Do not worry here.

**Comment by mchacki** [comment]
Jenkins:
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/8927/
The red tests are known in the base branch.
Will be fixed after merge

**Pull request merged**